addons/point_of_sale/static/src/js/models.js

```diff
@@ -1318,7 +1322,12 @@ exports.Orderline = Backbone.Model.extend({
         this.pos   = options.pos;
         this.order = options.order;
         if (options.json) {
-            this.init_from_JSON(options.json);
+            try {
+                this.init_from_JSON(options.json);
+            } catch(error) {
+                console.error('ERROR: attempting to recover product ID', json.product_id,
+                    'not available in the point of sale. Correct the product or clean the browser cache.');
+            }
```

Comment:
Maybe log `error` as well or is it not meaningful?

Reply:
the error is not meaningful in this case. I'm going to leave it like this but I am working on another task to handle this issue in a better way. 

https://www.odoo.com/web#id=1879971&action=333&active_id=1428&model=project.task&view_type=form&menu_id=4720